tracks/julia/exercises/word-count/mentoring.md

```diff
@@ -121,4 +121,94 @@ function wordcount(sentence)
 end
 ```
 
+** Without Regex ** 
+
+This solution doesn't use a regex at all. It uses the default split behavior (which checks for `isspace` characters) and then strips punctuation characters left over on the "outside" of words. Filtering hoas to come after stripping in case the strip operation itself left an empty string.
+```julia
+function wordcount(sentence)
+    words = strip.(ispunct, split(lowercase(sentence)))
+    filter!(!isempty, words)
+    d = Dict{eltype(words), Int}()
+    for w in words
+        d[w] = get(d, w, 0) + 1
+    end
+    d
+end
+```
+It's possible to create the dictionary more efficiently with the `countmap` function from `StatsBase` (which only hashes the key once per check). With countmap, you can code-golf the above into:
+```julia
+wordcount_withstatsbase(sentence) = countmap(filter!(!isempty, strip.(ispunct, split(lowercase(sentence)))))
+```
+
+
+** Totally manual ** 
+
+This method iterates through the string one character at a time and switches behavior at word boundaries (when `iswordchar` changes from true to false or false to true). The dictionary has `SubString` keys in this version, which are like the string-version of Array `views`.
+```julia
+function wordcount(sent)
+
+    sent = lowercase(sent)
+    d = Dict{SubString, Int}()
+
+    in_word = false; j = 0
+    # For our purposes, almost everything is a valid word character.
+    iswordchar(c) = !ispunct(c) && !isspace(c)
+    for i in eachindex(sent)
+        c = sent[i]
+        if iswordchar(c) && i < lastindex(sent)
+            # if we're not already in a word, then this character marks
+            # the start of a new word. If we are in a word, just keep going.
+            if !in_word
+                in_word = true
+                j = i
+            end
+        elseif in_word
+            # If this character is not a word character, but we are in a word,
+            # then the previous character marks the end of a word. However, if the
+            # current character is a ' we need to see if this word is a contraction
+            if c == '''
+                i < lastindex(sent) && iswordchar(sent[nextind(sent, i)]) && continue
+            end
+
+            in_word = false
+            k = SubString(sent, j:(i-1))
+            d[k] = get(d, k, 0) + 1
+        end
+    end
+    d
+end
+```
+
+### Benchmark comparison in julia v1.5.1
```

Comment:
Not sure about including benchmarking code. If we do, we should use bold for headers and make it reproducible with the earlier examples (i.e. they should be renamed wordcount1, wordcount2, etc).

Reply:
Yes, fair enough. What is your preference here, omit the benchmark, or rename the functions?

Reply:
For the record, I said this on slack:

> I think we should just omit it. Users can run their own benchmarks.
> If solutions have substantially different run times we should report that in the prose above the relevant solution.